Provider with HTTPS defaults fails with Invalid config.SSLCertPEM

The default value for the `ssl_cert_pem` argument results in an invalid `SSLCertPem` configuration value, rather than omitting it to use the default CA store for validation.

```
2018-02-06T19:35:20.714+0200 [DEBUG] plugin.terraform-provider-kontena: 2018/02/06 19:35:20 [DEBUG] config client.Config{URL:"https://wispy-sun-3205.platforms.us-east-1.kontena.cloud", SSLCertPEM:[]uint8{}, SSLServerName:"", ClientID:"", ClientSecret:"", Token:(*client.Token)(nil), Logger:(*kontena.Logger)(0x139d720)}
2018-02-06T19:35:20.714+0200 [DEBUG] plugin.terraform-provider-kontena: 2018/02/06 19:35:20 [DEBUG] connect https://wispy-sun-3205.platforms.us-east-1.kontena.cloud (token &{2c6778fa27a379fc2f020bb3b99b3f621e8e540fe4ce5389bda0267ce6ebac9e   0001-01-01 00:00:00 +0000 UTC <nil>})
2018/02/06 19:35:20 [ERROR] root: eval: *terraform.EvalConfigProvider, err: Invalid config.SSLCertPEM
2018/02/06 19:35:20 [ERROR] root: eval: *terraform.EvalSequence, err: Invalid config.SSLCertPEM
2018/02/06 19:35:20 [ERROR] root: eval: *terraform.EvalOpFilter, err: Invalid config.SSLCertPEM
2018/02/06 19:35:20 [ERROR] root: eval: *terraform.EvalSequence, err: Invalid config.SSLCertPEM
2018/02/06 19:35:20 [TRACE] [walkImport] Exiting eval tree: provider.kontena
2018/02/06 19:35:20 [TRACE] dag/walk: upstream errored, not walking "kontena_grid.demo (import id: demo-3)"
2018/02/06 19:35:20 [TRACE] dag/walk: upstream errored, not walking "kontena_grid.demo"
2018/02/06 19:35:20 [TRACE] dag/walk: upstream errored, not walking "kontena_external_registry.test"
2018/02/06 19:35:20 [TRACE] dag/walk: upstream errored, not walking "provider.kontena (close)"

Error: provider.kontena: Invalid config.SSLCertPEM

```

--- a/kontena/provider.go
+++ b/kontena/provider.go
@@ -70,10 +70,15 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		logger: &logger,
 		config: client.Config{
 			URL:           d.Get("url").(string),
-			SSLCertPEM:    []byte(d.Get("ssl_cert_pem").(string)),
-			SSLServerName: d.Get("ssl_cert_cn").(string),
 			Logger:        &logger,
 		},
+	}
+
+	if sslCertPEM, ok := d.GetOk("ssl_cert_pem"); ok {
+		meta.config.SSLCertPEM = []byte(sslCertPEM.(string))
+	}
+	if sslServerName, ok := d.GetOk("ssl_cert_cn"); ok {
+		meta.config.SSLServerName = sslServerName.(string)
 	}
 
 	logger.Debugf("config %#v", meta.config)

